src/App.js

```diff
@@ -2,19 +2,55 @@ import React, { Component } from 'react';
 import './App.css';
 
 class App extends Component {
-  state = { blocks: ['hello world', 'another string'] };
+  state = {
+    blocks: [
+      { type: 'heading', text: 'hello world' },
+      { type: 'text', text: 'another string' },
+      { type: 'image', url: 'http://fpoimg.com/300x300', text: 'FPO image' },
+      { type: 'blockQuote', text: 'this is a quote' },
```

Comment:
Let's actually move the supported types out into their own object so we don't need to worry about formatting and upper/lower case when we type them here and in the switch … so, outside this class create a new `const` like this:

```js
const BlockType = {
  Heading: 'heading',
  Text: 'text',
  Image: 'image',
  BlockQuote: 'blockQuote',
};

// then later, use this object instead of the literal strings, like:

blocks: [
  { type: BlockType.Heading, text: 'hello world' },
  // etc...
],

// and in the switch like:

switch (block.type) {
  case BlockType.Heading:
  // etc...
```

This just makes it a little less likely that a typo will trip you up later. At some point we can put this `BlockType` object in it's own file and share it amongst multiple components. Plus you should get "code hinting" when you type `BlockType.` a little pop up should show you the valid options.